### PR TITLE
refactor: change the signature of `Provider#complete`

### PR DESCRIPTION
**Pull request by antaz**
- Changed `Provider#complete` signature to take a `Message` parameter rather than separate prompt and role.
- Inlining `PATH` for `Provider#complete`.

### CODE REVIEW

File: lib/llm/provider.rb [hunk 1]
```diff
@@ -41,7 +41,7 @@ def complete(prompt, role = :user, **params)
     # @raise (see LLM::Provider#complete)
     # @return [LLM::Conversation]
     def chat(prompt, role = :user, **params)
-      completion = complete(prompt, role, **params)
+      completion = complete(Message.new(role, prompt), **params)
```
Comment (0x1eef) [inline lib/llm/provider.rb:44]: @antaz We could call `transform_prompt(prompt)` here.

File: lib/llm/providers/gemini.rb [hunk 2]
```diff
@@ -18,19 +18,19 @@ def initialize(secret)
     end
 
     def embed(input, **params)
-      path = [PATH, "text-embedding-004"].join("/")
```
Comment (0x1eef) [inline lib/llm/providers/gemini.rb:21]: @antaz Did we forget to delete the _assignment of the constant_ from this file ?

**Review by 0x1eef — APPROVED** [review]
Nice work :+1:

File: lib/llm/providers/gemini.rb (hunk 2 above)
Comment (antaz) [reply lib/llm/providers/gemini.rb:21]: @0x1eef We certainly did! Thank you

**Pull request merged**